ballerina/listener_utils.bal

```diff
@@ -102,6 +106,176 @@ isolated function addDefaultDirectives(__Schema schema) {
     }
 }
 
+isolated function getResponseFromMultipartPayload(Engine engine, Context context, http:Request request)
+    returns http:Response {
+    map<FileUpload> fileInfo = {};
+    map<json> pathMap = {};
+    map<json> variables = {};
+    json payload = ();
+    map<FileUpload|FileUpload[]> files = {};
+    mime:Entity[]|http:ClientError bodyParts = request.getBodyParts();
+    if bodyParts is mime:Entity[] {
+        foreach mime:Entity part in bodyParts {
+            mime:ContentDisposition contentDisposition = part.getContentDisposition();
+            if contentDisposition.name == MULTIPART_OPERATIONS {
+                json|mime:ParserError operation = part.getJson();
+                if operation is json {
+                    var variableValues = operation.variables;
+                    payload = operation;
+                    variableValues = variableValues is error ? () : variableValues;
+                    if variableValues is map<json> {
+                        variables = variableValues;
+                    } else {
+                        return createResponse("Invlaid Mulitpart Request", http:STATUS_BAD_REQUEST);
+                    }
+                } else {
+                    return createResponse(operation.message(), http:STATUS_BAD_REQUEST);
+                }
+            } else if contentDisposition.name == MULITPART_MAP {
+                json|mime:ParserError paths = part.getJson();
+                if paths is json {
```

Comment:
This cast can cause issues, right? Shouldn't the type guard should be the following?
```suggestion
                if paths is map<json> {
```